libr/util/str.c

```diff
@@ -2928,3 +2929,21 @@ R_API char* r_str_highlight(char *str, const char *word, const char *color) {
 	}
 	return strdup (o);
 }
+
+R_API int r_str_scpy (char *dst, const char *src, int count) {
+  if (count <= 0) {
+    return -E2BIG;
+  }
+  int ret = 0;
+  while (count) {
+    if (!(dst[ret] = src[ret])) {
+      return ret;
+    }
+    ret++;
+    count--;
+  }
+  if (ret) {
+    dst[ret-1] = '\0';
+  }
+  return -E2BIG;
```

Comment:
in strlcpy manpage:

```

RETURN VALUES
     The strlcpy() and strlcat() functions return the total length of the string they tried to create.  For strlcpy() that means the length of src.  For strlcat() that means the initial
     length of dst plus the length of src.  While this may seem somewhat confusing, it was done to make truncation detection simple.

     Note however, that if strlcat() traverses size characters without finding a NUL, the length of the string is considered to be size and the destination string will not be NUL-termi-
     nated (since there was no space for the NUL).  This keeps strlcat() from running off the end of a string.  In practice this should not happen (as it means that either size is incor-
     rect or that dst is not a proper ``C'' string).  The check exists to prevent potential security problems in incorrect code.
```